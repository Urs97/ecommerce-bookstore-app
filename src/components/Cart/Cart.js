import React, { useEffect, useRef } from 'react';
import './Cart.scss';
import PageHeader from '../PageHeader/PageHeader';
import CartItem from '../CartItem/CartItem';
import CartCheckout from '../CartCheckout/CartCheckout';

const Cart = () => {
    const CartCheckoutRef = useRef(null);

    useEffect(() => {
        window.scrollTo(0, 0);
      }, [])

    const scrollToRef = (ref) => window.scrollTo(0, ref.current.offsetTop, 'smooth');
    const executeScroll = () => scrollToRef(CartCheckoutRef);

    return (
        <main>
            <PageHeader title="Your Cart" color="green"/>
            <section className="cart-container">
                <section className="cart-nav">
                    <h1>Shopping Cart- total- nr of items</h1>
                    <section className="cart-nav-btns">
                        <a href="/store" className="button continue-btn">Continue Shopping</a>
                        <button className="button checkout-btn"
                                onClick={() => executeScroll()}>To Checkout</button>
                    </section>
                </section>
                <section className="product-container">
                    <CartItem header={"header"}/>
                    <CartItem />
                    <CartItem />
                    <CartItem />
                    <CartItem />
                </section>
            </section>
            <section ref={CartCheckoutRef}>
                <CartCheckout />
            </section>
        </main>
    )
};

export default Cart;
