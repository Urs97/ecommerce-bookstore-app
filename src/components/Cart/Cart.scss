@import '../../styles/shared.scss';

.cart-container {
    padding: $vertical-content-padding $horizontal-content-padding;
    padding-bottom: 3em;
    background-color: #F4F7F9;    

    .cart-nav {
        padding-bottom: 2em;
        display: flex;
        align-items: center;
        justify-content: space-between;

        .cart-nav-btns {

            .continue-btn {
                background-color: $dark-blue;
            }
    
            .continue-btn:hover {
                background-color: $yellow;
            }
    
            .checkout-btn {
                margin-left: 1em;
                background-color: red;
            }
    
            .checkout-btn:hover {
                background-color: $yellow;
            }
        }
    }

    .product-container {
        background-color: #FFFFFF;
        border-radius: $border-radius $border-radius 0 0;
        box-shadow: 0 0 20px rgba(0, 0, 0, 0.3);
    }

    .product-container > *:not(:last-child) {
        border-bottom: 0.1px solid $green;
    }

    .product-container:last-child {
        border-bottom: 0.2em solid $green;
    }
}
