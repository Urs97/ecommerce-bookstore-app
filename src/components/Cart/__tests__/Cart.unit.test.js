import React from "react";
import { shallow } from "enzyme";
import Cart from "../Cart";

describe("Cart", () => {
  test("renders Cart", () => {
    const wrapper = shallow(<Cart />);
    expect(wrapper).toMatchSnapshot();
  });
});
