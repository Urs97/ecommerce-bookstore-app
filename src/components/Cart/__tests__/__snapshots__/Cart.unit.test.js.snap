// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart renders Cart 1`] = `
<main>
  <PageHeader
    color="green"
    title="Your Cart"
  />
  <section
    className="cart-container"
  >
    <section
      className="cart-nav"
    >
      <section
        className="cart-nav-title"
      >
        <h1
          className="title"
        >
          Shopping Cart
        </h1>
        <h4
          className="title"
        >
          $
          0.00
        </h4>
        <h4
          className="title"
        >
          0
           items
        </h4>
      </section>
      <section
        className="cart-nav-btns"
      >
        <Link
          className="button continue-btn"
          to="/store"
        >
          Continue Shopping
        </Link>
      </section>
    </section>
    <section
      className="product-container"
    >
      <section
        className="empty-cart"
      >
        <span>
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  576,
                  512,
                  Array [],
                  "f07a",
                  "M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z",
                ],
                "iconName": "shopping-cart",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
        </span>
        <h1
          className="title"
        >
          Your Cart Is Currently Empty!
        </h1>
      </section>
    </section>
  </section>
</main>
`;
